db/disk/course.go

```diff
@@ -59,12 +65,15 @@ func (this *backend) SaveCourse(course *model.Course) error {
 }
 
 func (this *backend) saveCourseLock(course *model.Course, acquireLock bool) error {
+    courseDir := this.getCourseDir(course);
+    courseDir = util.ShouldAbs(courseDir);
+
     if (acquireLock) {
-        this.lock.Lock();
-        defer this.lock.Unlock();
+        common.Lock(courseDir);
+        defer common.Unlock(courseDir);
     }
 
-    util.MkDir(this.getCourseDir(course));
+    util.MkDir(courseDir);
```

Comment:
Careful because of the above comment.
Make sure this gets changed when the above comment gets resolved.